### DISCUSSION TITLE
Show message ID on invalid attachment?

### DISCUSSION THREAD

**Discussion by spiffzilla** [GAM-team/got-your-back#348] [Q&A]
Is it possible to show message ID on invalid attachment when doing a restore?

Would be nice to see which messages that is excluded from the restote, tried the --debug flag but coldn't find anything there

ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.

//Jörgen

**Comment by spiffzilla** [comment]
Didn't see anything in the DB that indicated if a message was copied/invalid or not...

**Reply by flipflophhj** [reply]
I have the same problem. Got all my mails from my GSuite restored to my regular old gmail except 36 messages with invalid attachment.

**Reply by spiffzilla** [reply]
from my limited knowledge in python my guess is that here would be a good place to print which message that had an error:

(line 1547 and forward in  gyb.py ) 
def restored_message(request_id, response, exception):
  if exception is not None:
    try:
      error = json.loads(exception.content.decode('utf-8'))
      if error['error']['code'] == 400:
        print("\nERROR: %s: %s. Skipping message restore."
          % (error['error']['code'], error['error']['errors'][0]['message']))
        return

**Comment by dab15** [comment]
I am migrating out of a GSuite account and I am getting that error a *lot*!  I would love to know which messages were affected. Out of 93,000+ messages I am migrating, I am unlikely to have time to go back and figure this out.

 ERROR: 400: Precondition check failed.. Skipping message restore.
restoring 1.17mb message (9196/82655)                                           
400: Precondition check failed. - failedPrecondition
 - Giving up.
restoring 15 messages (10141/82655)                                             
ERROR: 400: Precondition check failed.. Skipping message restore.
restoring 15 messages (12812/82655)                                             
ERROR: 400: Precondition check failed.. Skipping message restore.
restoring 15 messages (13254/82655)                                             
ERROR: 400: Precondition check failed.. Skipping message restore.
restoring 15 messages (14610/82655)                                             
ERROR: 400: Precondition check failed.. Skipping message restore.
restoring 15 messages (14701/82655)                                             
ERROR: 400: Precondition check failed.. Skipping message restore.
restoring 15 messages (15291/82655)

**Reply by alaflaquiere** [reply]
I have facing a smiliar problem.
Uploading some mbox to a gmail account, but 3 messages raise the "ERROR: 400: Invalid attachment" error.
I would love to find a way to identify which emails are the source of the problem. Any tip is welcome!

**Reply by rjmccallumbigl** [reply]
I just ran into this doing a Google Takeout to get an MBOX of my gmail messages and importing them to another gmail. Figured this out in case this helps someone else:

1. rerun the script to get an output of the failed imports. When I reran `.\gyb.exe --action restore-mbox --local-folder "F:\imported from email\Al" --email .\gyb.exe --action restore-mbox --local-folder "F:\imported from email" --email myemailaddress@gmail.com` it looked like this which gave me the failed imports:

```
Restoring from 8.04gb file F:\imported from email\All mail Including Spam and Trash-002.mbox...
 restoring 6.1mb message 112621 - 80.15%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1.37mb message 137906 - 94.06%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1.34mb message 137907 - 94.07%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1 messages (141590) - 95.82%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 3.26mb message 144278 - 97.31%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1 messages (145269) - 97.91%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1 messages (145270) - 97.91%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1 messages (145386) - 97.96%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1 messages (145387) - 97.96%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 restoring 1.2mb message 148480 - 99.37%
ERROR: 400: Invalid attachment. Please check https://support.google.com/mail/answer/6590.. Skipping message restore.
 skipping already restored message #149547
done!
```

2. Then I reran with the debug log redirected to a text file:  `.\gyb.exe --action restore-mbox --local-folder "F:\imported from email\Al" --email .\gyb.exe --action restore-mbox --local-folder "F:\imported from email" --email myemailaddress@gmail.com --debug > debug.log` 

3. Opened `debug.log` in VS Code and CTRL + F the failed imports (in my case `112621` was my first one) and can see the subject of the email in the debug log between `112621` and `112622`

**Reply by rjmccallumbigl** [reply]
Also just noticed some of my debug log responses only have the raw encoding and not the subject of the email (looks like `"raw": "WC1HTS1USFJJRDog....`)

I can take the full raw base64 string and decode using a tool such as https://www.base64decode.org/ to view the text of the email and decide if it's important or not
